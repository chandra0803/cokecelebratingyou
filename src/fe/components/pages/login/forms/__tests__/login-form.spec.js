/*global jest*/

jest.mock( 'react-redux' );
// jest.mock( 'react-router-dom' );
// jest.mock( '../../../shared/locale-string' );
import { mount } from 'enzyme';

import LoaderButton from '../../../../shared/loader-button/loader-button';
import React from 'react';
import renderer from 'react-test-renderer';
import LoginForm from '../login-form';

    const defaultProps = {
            login: {

                locales: [

                        {
                            code: 'zh_CN',
                            label: 'Chinese [Simplified]'
                        },

                        {
                            code: 'zh_TW',
                            label: 'Chinese [Traditional]'
                        },

                        {
                            code: 'nl_NL',
                            label: 'Dutch [Netherlands]'
                        },

                        {
                            code: 'en_GB',
                            label: 'English [British]'
                        },

                        {
                            code: 'en_US',
                            label: 'English [U.S]'
                        },

                        {
                            code: 'fr_CA',
                            label: 'French [Canadian]'
                        },

                        {
                            code: 'fr_FR',
                            label: 'French [Europe]'
                        },

                        {
                            code: 'de_DE',
                            label: 'German [Germany]'
                        },

                        {
                            code: 'it_IT',
                            label: 'Italian [Italy]'
                        },

                        {
                            code: 'ja_JP',
                            label: 'Japanese [Japan]'
                        },

                        {
                            code: 'ar_AE',
                            label: 'Klingon'
                        },

                        {
                            code: 'ko_KR',
                            label: 'Korean [South Korea]'
                        },

                        {
                            code: 'pl_PL',
                            label: 'Polish [Poland]'
                        },

                        {
                            code: 'pt_BR',
                            label: 'Portuguese [Brazil]'
                        },

                        {
                            code: 'ru_RU',
                            label: 'Russian [Russia]'
                        },

                        {
                            code: 'es_MX',
                            label: 'Spanish [Latin America]'
                        },

                        {
                            code: 'es_ES',
                            label: 'Spanish [Spain]'
                        },

                        {
                            code: 'th_TH',
                            label: 'Thai [Thailand]'
                        },

                        {
                            code: 'tr_TR',
                            label: 'Turkish [Turkey]'
                        },

                        {
                            code: 'vi_VN',
                            label: 'Vietnamese [Vietnam]'
                        },

                ],

            },
            content: {
                'login.errors.login_id_req': 'Login ID is required.',
                'login.errors.password_req': 'Password is required.',
                'login.loginpage.activate_account': 'Activate your account',
                'login.loginpage.activation': 'Activation',
                'login.loginpage.activation_attributes': 'Activation Attributes',
                'login.loginpage.back_to_login': 'Back to Login',
                'login.loginpage.code_verification': 'Code Verification',
                'login.loginpage.confirm_password': 'Confirm Password',
                'login.loginpage.contact_information': 'Contact Information',
                'login.loginpage.country_code': 'Country Code',
                'login.loginpage.create_password': 'Create Password',
                'login.loginpage.forgot_login_id': 'Forgot Login ID',
                'login.loginpage.forgot_password': 'Forgot Password',
                'login.loginpage.help': 'Help',
                'login.loginpage.i_forgot_login_id': 'I forgot my login ID',
                'login.loginpage.i_forgot_password': 'I forgot my password',
                'login.loginpage.intro_para_one': 'This is the one-stop shop for your program. Check back often for updates!',
                'login.loginpage.log_in': 'Log In',
                'login.loginpage.message_sent': 'Message Sent',
                'login.loginpage.method_of_contact': 'Method of Contact',
                'login.loginpage.new_password': 'New Password',
                'login.loginpage.no_login_id_or_password': 'No Login ID or Password?',
                'login.loginpage.no_password': 'No Password?',
                'login.loginpage.one_time_code': 'One Time Code',
                'login.loginpage.password': 'Password',
                'login.loginpage.recovery_information': 'Recovery Information',
                'login.loginpage.register': 'Register',
                'login.loginpage.register_new_account': 'Register a new account',
                'login.loginpage.reset_password': 'Reset Password',
                'login.loginpage.title': 'Welcome',
                'login.loginpage.username': 'Login ID',
				'login.loginpage.user_id_is_your_employee_number': 'User id is your employee number',
                'login.password.reset.phone_number': 'Phone Number',
                'participant.participant.email_address': 'Email Address',
                'system.button.close': 'Close',
                'system.button.submit': 'Submit',
                'system.general.contact_us': 'Contact Us',
                'system.general.copyright_text': 'Copyright &copy;2017 BI WORLDWIDE&trade;. All rights reserved.',
                'system.general.help_text': 'Contact your program administrator for help.',
                'system.general.help_title': 'Help',
                'system.general.privacy_policy': 'Privacy Policy',
                'system.general.t&c': 'T&Cs' },
            selfEnrollment: false,
            contactUsEmailConfirmation: false,
            strutsToken: '570210087a24f665d94727351f1c6597'
    };

it( 'renders the dom like the snapshot when the props have default values.', () => {
    const props = {
        ...defaultProps,
        setLanguage: jest.fn()
    };
    const component = renderer.create(
        <div>
            <LoginForm { ...props } />
        </div>
    );

    const tree = component.toJSON();
    expect( tree ).toMatchSnapshot();
    } );
    it( 'renders the dom like the snapshot when the props have default values.', () => {
        const content = {
            content: {
                'login.errors.login_id_req': 'Login ID is required.',
                'login.errors.password_req': 'Password is required.',
                'login.loginpage.activate_account': 'Activate your account',
                'login.loginpage.activation': 'Activation',
                'login.loginpage.activation_attributes': 'Activation Attributes',
                'login.loginpage.back_to_login': 'Back to Login',
                'login.loginpage.code_verification': 'Code Verification',
                'login.loginpage.confirm_password': 'Confirm Password',
                'login.loginpage.contact_information': 'Contact Information',
                'login.loginpage.country_code': 'Country Code',
                'login.loginpage.create_password': 'Create Password',
                'login.loginpage.forgot_login_id': 'Forgot Login ID',
                'login.loginpage.forgot_password': 'Forgot Password',
                'login.loginpage.help': 'Help',
                'login.loginpage.i_forgot_login_id': 'I forgot my login ID',
                'login.loginpage.i_forgot_password': 'I forgot my password',
                'login.loginpage.intro_para_one': 'This is the one-stop shop for your program. Check back often for updates!',
                'login.loginpage.intro_para_two': 'paragraph 2!',
                'login.loginpage.log_in': 'Log In',
                'login.loginpage.message_sent': 'Message Sent',
                'login.loginpage.method_of_contact': 'Method of Contact',
                'login.loginpage.new_password': 'New Password',
                'login.loginpage.no_login_id_or_password': 'No Login ID or Password?',
                'login.loginpage.no_password': 'No Password?',
                'login.loginpage.one_time_code': 'One Time Code',
                'login.loginpage.password': 'Password',
                'login.loginpage.recovery_information': 'Recovery Information',
                'login.loginpage.register': 'Register',
                'login.loginpage.register_new_account': 'Register a new account',
                'login.loginpage.reset_password': 'Reset Password',
                'login.loginpage.title': 'Welcome',
                'login.loginpage.username': 'Login ID',
				'login.loginpage.user_id_is_your_employee_number': 'User id is your employee number',
                'login.password.reset.phone_number': 'Phone Number',
                'participant.participant.email_address': 'Email Address',
                'system.button.close': 'Close',
                'system.button.submit': 'Submit',
                'system.general.contact_us': 'Contact Us',
                'system.general.copyright_text': 'Copyright &copy;2017 BI WORLDWIDE&trade;. All rights reserved.',
                'system.general.help_text': 'Contact your program administrator for help.',
                'system.general.help_title': 'Help',
                'system.general.privacy_policy': 'Privacy Policy',
                'system.general.t&c': 'T&Cs' },
        };
        const login = {
            login: {
                locales: []
            }
        };
        const props = {
            ...content,
            ...login,
            setLanguage: jest.fn()
        };
        const component = renderer.create(
            <div>
                <LoginForm { ...props } />
            </div>
        );

        const tree = component.toJSON();
        expect( tree ).toMatchSnapshot();
        } );
it ( 'disables the button when username and password have not been entered', () => {
    const props = {
        ...defaultProps,
        setLanguage: jest.fn()
    };
    const tree = mount( <LoginForm { ...props } /> );

    const button = tree.find( LoaderButton );

    expect ( button.props().disabled ).toBe( true );

    tree.setState( { username: 'testing' } );
    expect ( button.props().disabled ).toBe( true );

    tree.setState( { username: '', password: 'testing' } );
    expect ( button.props().disabled ).toBe( true );


    tree.setState( { username: 'testing', password: 'testing' } );
    expect ( button.props().disabled ).toBe( false );

} );
it ( 'executes changeForm when the Forgot Login ID link is clicked.', () => {
    const props = {
        ...defaultProps,
        handleFormChange: jest.fn(),
        setLanguage: jest.fn()
    };

    const tree = mount( <LoginForm { ...props } /> );

    const link = tree.find( '.forgot-link' ).first();

    link.simulate( 'click' );

    expect ( props.handleFormChange.mock.calls.length ).toBe( 1 );
} );
it ( 'executes changeForm when the Forgot Password ID link is clicked.', () => {
    const props = {
        ...defaultProps,
        handleFormChange: jest.fn(),
        setLanguage: jest.fn()
    };

    const tree = mount( <LoginForm { ...props } /> );

    const link = tree.find( '.forgot-link' ).at( 1 );

    link.simulate( 'click' );

    expect ( props.handleFormChange.mock.calls.length ).toBe( 1 );
} );
it ( 'executes changeForm when the activate account link is clicked.', () => {
    const props = {
        ...defaultProps,
        handleFormChange: jest.fn(),
        setLanguage: jest.fn()
    };

    const tree = mount( <LoginForm { ...props } /> );

    const link = tree.find( 'a' ).at( 2 );

    link.simulate( 'click' );

    expect ( props.handleFormChange.mock.calls.length ).toBe( 1 );
} );
it ( 'executes togglePasswordVisible when the eye icon is clicked.', () => {
    const props = {
        ...defaultProps,
        togglePasswordVisible: jest.fn(),
        setLanguage: jest.fn()
    };

    const tree = mount( <LoginForm { ...props } /> );

    const link = tree.find( '.eye' );

    link.simulate( 'click' );

    expect ( props.togglePasswordVisible.mock.calls.length ).toBe( 1 );
} );
it ( 'executes setLanguage and changeLanguage when the language select is changed.', () => {
    const props = {
        ...defaultProps,
        setLanguage: jest.fn(),
        changeLanguage: jest.fn()
    };

    const tree = mount( <LoginForm { ...props } /> );

    const select = tree.find( 'select' );

    select.simulate( 'change', { target: { value: 'en_GB' } } );

    expect ( props.setLanguage.mock.calls.length ).toBe( 1 );
    expect ( props.changeLanguage.mock.calls.length ).toBe( 1 );
} );
it ( 'executes loginFunc prop when enter is pressed.', () => {
    const props = {
        ...defaultProps,
        setLanguage: jest.fn(),
        loginFunc: jest.fn()
    };

    const tree = mount( <LoginForm { ...props } /> );

    const input = tree.find( 'input[name="username"]' );

    tree.setState( { username: 'testing', password: 'testing' } );

    input.simulate( 'keyPress', { charCode: 13 } );


    expect ( props.loginFunc.mock.calls.length ).toBe( 1 );
} );
it ( 'does not execute loginFunc prop when enter is pressed with empty inputs.', () => {
    const props = {
        ...defaultProps,
        setLanguage: jest.fn(),
        loginFunc: jest.fn()
    };

    const tree = mount( <LoginForm { ...props } /> );

    const input = tree.find( 'input[name="username"]' );

    tree.setState( { username: '', password: '' } );

    input.simulate( 'keyPress', { charCode: 13 } );

    expect ( props.loginFunc.mock.calls.length ).toBe( 0 );
    const password = tree.find( 'input[name="password"]' );
    password.simulate( 'keyPress', { charCode: 13 } );
    expect ( props.loginFunc.mock.calls.length ).toBe( 0 );
} );
it ( 'changes username in state when the user changes something in the username input ', () => {
    const props = {
        ...defaultProps,
        setLanguage: jest.fn()
    };

    const tree = mount( <LoginForm { ...props } /> );

    const input = tree.find( 'input[name="username"]' );

    input.simulate( 'change', { target: { name: input.props().name, value: 'test' } } );

    expect ( tree.state( 'username' ) ).toBe( 'test' );
} );
it ( 'changes password in state when the user changes something in the password input ', () => {
    const props = {
        ...defaultProps,
        setLanguage: jest.fn()
    };

    const tree = mount( <LoginForm { ...props } /> );

    const input = tree.find( 'input[name="password"]' );

    input.simulate( 'change', { target: { name: input.props().name, value: 'test' } } );

    expect ( tree.state( 'password' ) ).toBe( 'test' );
} );
it ( 'shows a error alert when there is a error message.', () => {
    const login = {
        login: {
            ...defaultProps.login,
            message: {
                error: 'error'
            }
        }
    };
    const props = {
        ...defaultProps,
        setLanguage: jest.fn(),
        ...login
    };

    const tree = mount( <LoginForm { ...props } /> );

    const alerts = tree.find( '.alert-error' );

    expect ( alerts.length ).toBe( 1 );
} );
it ( 'shows an  alert when there is a server message.', () => {
    const login = {
        login: {
            ...defaultProps.login,
            message: {
                type: 'success',
                name: '',
                id: '',
                text: ''
            }
        }
    };
    const props = {
        ...defaultProps,
        setLanguage: jest.fn(),
        ...login
    };

    const tree = mount( <LoginForm { ...props } /> );

    const alerts = tree.find( '.alert-success' );

    expect ( alerts.length ).toBe( 1 );
} );
it ( 'shows a success alert when there is a success message.', () => {
    const login = {
        login: {
            ...defaultProps.login,
            message: {
                success: 'success'
            }
        }
    };
    const props = {
        ...defaultProps,
        setLanguage: jest.fn(),
        ...login
    };

    const tree = mount( <LoginForm { ...props } /> );

    const alerts = tree.find( '.alert-success' );

    expect ( alerts.length ).toBe( 1 );
} );
it ( 'shows an eye open icon when passwordVisible is true.', () => {
    const login = {
        login: {
            ...defaultProps.login,
            passwordVisible: true
        }
    };
    const props = {
        ...defaultProps,
        setLanguage: jest.fn(),
        ...login
    };

    const tree = mount( <LoginForm { ...props } /> );

    const eye = tree.find( '.eye' );
    expect ( eye.length ).toBe( 1 );
} );
it ( 'executes toggleModal when the register account link is clicked.', () => {
    const login = {
        login: {
            ...defaultProps.login,
            selfEnrollment: true
        }
    };
    const props = {
        ...defaultProps,
        ...login,
        toggleModalDisplay: jest.fn(),
        setLanguage: jest.fn()
    };

    const tree = mount( <LoginForm { ...props } /> );

    const link = tree.find( 'a' ).at( 3 );

    link.simulate( 'click' );

    expect ( props.toggleModalDisplay.mock.calls.length ).toBe( 1 );
} );
