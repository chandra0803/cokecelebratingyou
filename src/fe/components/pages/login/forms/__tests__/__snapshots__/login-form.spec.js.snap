// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the dom like the snapshot when the props have default values. 1`] = `
<div>
  <div
    className="narrow-form-page login"
  >
    <form
      action={undefined}
      className="narrow-form"
      method="post"
    >
      <div
        className="row-fluid"
      >
        <h2>
          Welcome
        </h2>
        <p>
          This is the one-stop shop for your program. Check back often for updates!
        </p>
        <div
          className="language-select input-wrap"
        >
          <label
            className="control-label"
            htmlFor="languageSelect"
          />
          <select
            className="selectpicker gq-input"
            id="language-select"
            onChange={[Function]}
            value={undefined}
          >
            <option
              value="zh_CN"
            >
              Chinese [Simplified]
            </option>
            <option
              value="zh_TW"
            >
              Chinese [Traditional]
            </option>
            <option
              value="nl_NL"
            >
              Dutch [Netherlands]
            </option>
            <option
              value="en_GB"
            >
              English [British]
            </option>
            <option
              value="en_US"
            >
              English [U.S]
            </option>
            <option
              value="fr_CA"
            >
              French [Canadian]
            </option>
            <option
              value="fr_FR"
            >
              French [Europe]
            </option>
            <option
              value="de_DE"
            >
              German [Germany]
            </option>
            <option
              value="it_IT"
            >
              Italian [Italy]
            </option>
            <option
              value="ja_JP"
            >
              Japanese [Japan]
            </option>
            <option
              value="ar_AE"
            >
              Klingon
            </option>
            <option
              value="ko_KR"
            >
              Korean [South Korea]
            </option>
            <option
              value="pl_PL"
            >
              Polish [Poland]
            </option>
            <option
              value="pt_BR"
            >
              Portuguese [Brazil]
            </option>
            <option
              value="ru_RU"
            >
              Russian [Russia]
            </option>
            <option
              value="es_MX"
            >
              Spanish [Latin America]
            </option>
            <option
              value="es_ES"
            >
              Spanish [Spain]
            </option>
            <option
              value="th_TH"
            >
              Thai [Thailand]
            </option>
            <option
              value="tr_TR"
            >
              Turkish [Turkey]
            </option>
            <option
              value="vi_VN"
            >
              Vietnamese [Vietnam]
            </option>
          </select>
        </div>
        <div
          className="username-field login-form input-wrap focused"
        >
          <a
            className="forgot-link login-input-label"
            onClick={[Function]}
            tabIndex="5"
          >
            <span>
              login.password.reset.forgot_login_id
            </span>
          </a>
          <label
            className="control-label"
            htmlFor="formUserId"
          >
            <span>
              Login ID
            </span>
          </label>
          <span
            className="userId"
          >
            <span
              className="icon-info"
            />
            <span
              className="instructions"
            />
          </span>
          <input
            autoFocus={true}
            className="gq-input"
            id="formUserId"
            name="username"
            onChange={[Function]}
            onKeyPress={[Function]}
            tabIndex="1"
            type="text"
            value=""
          />
        </div>
        <div
          className="password-field input-wrap focused"
        >
          <a
            className="forgot-link login-input-label"
            onClick={[Function]}
            tabIndex="5"
          >
            <span>
              Forgot Password
            </span>
          </a>
          <label
            className="control-label"
            htmlFor="formPassword"
          >
            <span>
              Password
            </span>
          </label>
          <span
            className="eye"
            onClick={[Function]}
          >
            <svg
              className="icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M20.6,6.8l-0.9,1.1c1.9,1.5,2.9,3.6,2.9,4.1c0,1.3-3.4,6.7-10.6,6.7c-1.1,0-2.5-0.2-3.4-0.4l-0.3,1.4 c0.9,0.2,2.5,0.4,3.6,0.4c7.8,0,12-5.8,12-8.1C24,10.8,22.6,8.4,20.6,6.8z"
              />
              <path
                d="M21.6,1.4L18,5c-1.4-0.8-4-1.1-6-1.1C4.2,3.9,0,9.6,0,12c0,1.2,1.6,4.3,4.6,6.4l-3.3,3.3l1,1L22.6,2.4L21.6,1.4z M1.4,12 c0-1.3,3.4-6.7,10.6-6.7c1.6,0,3.7,0.3,4.9,0.8L15,8c-0.9-0.6-1.9-1-3-1c-2.8,0-5,2.2-5,5c0,1.1,0.4,2.1,1,3l-2.3,2.3 C2.7,15.5,1.4,12.6,1.4,12z M8.4,12c0-2,1.6-3.6,3.6-3.6c0.7,0,1.4,0.2,2,0.6l-5,5C8.6,13.4,8.4,12.7,8.4,12z"
              />
              <path
                d="M12,15.6c-0.2,0-0.3,0-0.5,0l-0.2,1.4c0.2,0,0.4,0,0.7,0c2.8,0,5-2.2,5-5c0-0.2,0-0.4,0-0.7l-1.4,0.2c0,0.2,0,0.3,0,0.5 C15.6,14,14,15.6,12,15.6z"
              />
            </svg>
          </span>
          <input
            className="gq-input"
            id="formPassword"
            name="password"
            onChange={[Function]}
            onKeyPress={[Function]}
            tabIndex="2"
            type="password"
            value=""
          />
        </div>
        <div
          className="activation"
        >
          <p>
            No Password?
             
            <a
              onClick={[Function]}
            >
              Activate your account
            </a>
          </p>
        </div>
        <button
          className="gq-btn disabled btn btn-block btn-primary form-action-btn"
          onClick={[Function]}
          tabIndex={3}
          type="button"
        >
          <span>
            Log In
          </span>
        </button>
      </div>
    </form>
    <div
      className="theme-image"
    />
  </div>
</div>
`;

exports[`renders the dom like the snapshot when the props have default values. 2`] = `
<div>
  <div
    className="narrow-form-page login"
  >
    <form
      action={undefined}
      className="narrow-form"
      method="post"
    >
      <div
        className="row-fluid"
      >
        <h2>
          Welcome
        </h2>
        <p>
          This is the one-stop shop for your program. Check back often for updates!
        </p>
        <p>
          paragraph 2!
        </p>
        <div
          className="hide language-select input-wrap"
        >
          <label
            className="control-label"
            htmlFor="languageSelect"
          />
          <select
            className="selectpicker gq-input"
            id="language-select"
            onChange={[Function]}
            value={undefined}
          />
        </div>
        <div
          className="username-field login-form input-wrap focused"
        >
          <a
            className="forgot-link login-input-label"
            onClick={[Function]}
            tabIndex="5"
          >
            <span>
              login.password.reset.forgot_login_id
            </span>
          </a>
          <label
            className="control-label"
            htmlFor="formUserId"
          >
            <span>
              Login ID
            </span>
          </label>
          <span
            className="userId"
          >
            <span
              className="icon-info"
            />
            <span
              className="instructions"
            />
          </span>
          <input
            autoFocus={true}
            className="gq-input"
            id="formUserId"
            name="username"
            onChange={[Function]}
            onKeyPress={[Function]}
            tabIndex="1"
            type="text"
            value=""
          />
        </div>
        <div
          className="password-field input-wrap focused"
        >
          <a
            className="forgot-link login-input-label"
            onClick={[Function]}
            tabIndex="5"
          >
            <span>
              Forgot Password
            </span>
          </a>
          <label
            className="control-label"
            htmlFor="formPassword"
          >
            <span>
              Password
            </span>
          </label>
          <span
            className="eye"
            onClick={[Function]}
          >
            <svg
              className="icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M20.6,6.8l-0.9,1.1c1.9,1.5,2.9,3.6,2.9,4.1c0,1.3-3.4,6.7-10.6,6.7c-1.1,0-2.5-0.2-3.4-0.4l-0.3,1.4 c0.9,0.2,2.5,0.4,3.6,0.4c7.8,0,12-5.8,12-8.1C24,10.8,22.6,8.4,20.6,6.8z"
              />
              <path
                d="M21.6,1.4L18,5c-1.4-0.8-4-1.1-6-1.1C4.2,3.9,0,9.6,0,12c0,1.2,1.6,4.3,4.6,6.4l-3.3,3.3l1,1L22.6,2.4L21.6,1.4z M1.4,12 c0-1.3,3.4-6.7,10.6-6.7c1.6,0,3.7,0.3,4.9,0.8L15,8c-0.9-0.6-1.9-1-3-1c-2.8,0-5,2.2-5,5c0,1.1,0.4,2.1,1,3l-2.3,2.3 C2.7,15.5,1.4,12.6,1.4,12z M8.4,12c0-2,1.6-3.6,3.6-3.6c0.7,0,1.4,0.2,2,0.6l-5,5C8.6,13.4,8.4,12.7,8.4,12z"
              />
              <path
                d="M12,15.6c-0.2,0-0.3,0-0.5,0l-0.2,1.4c0.2,0,0.4,0,0.7,0c2.8,0,5-2.2,5-5c0-0.2,0-0.4,0-0.7l-1.4,0.2c0,0.2,0,0.3,0,0.5 C15.6,14,14,15.6,12,15.6z"
              />
            </svg>
          </span>
          <input
            className="gq-input"
            id="formPassword"
            name="password"
            onChange={[Function]}
            onKeyPress={[Function]}
            tabIndex="2"
            type="password"
            value=""
          />
        </div>
        <div
          className="activation"
        >
          <p>
            No Password?
             
            <a
              onClick={[Function]}
            >
              Activate your account
            </a>
          </p>
        </div>
        <button
          className="gq-btn disabled btn btn-block btn-primary form-action-btn"
          onClick={[Function]}
          tabIndex={3}
          type="button"
        >
          <span>
            Log In
          </span>
        </button>
      </div>
    </form>
    <div
      className="theme-image"
    />
  </div>
</div>
`;
